train: add `--forcing-data-path` CLI option to load forcing data directly instead of through MLflow

raehik opened the pull request:
Fixes #78 .

Should be backwards compatible with `--run-id` invocations. Note that we remove `--exp-id` due to it not being used (other than to for logging to screen).

raehik commented:
Note that this doesn't touch the rest of the MLflow integration: we still (optionally) execute the steps with MLflow, and save files using `mlflow.log_artifact()` in the data step.

MarionBWeinzierl commented on src/gz21_ocean_momentum/trainScript.py:
```diff
@@ -161,13 +168,35 @@ def check_str_is_None(string_in: str):
+    Returns a filepath which should be a zarr dataset. Correctness is not
+    asserted, so the filepath may not exist or may not be a valid zarr dataset.
+
+    TODO bit wonky the way we do mutual exclusion.
```
Maybe better put todo into issue and remove here

raehik replied:
Clarified with comments. I think it's fairly clear now. (I'm disappointed that I can't find a good way to do mutually exclusive arguments directly in argparse!)

MarionBWeinzierl commented on src/gz21_ocean_momentum/trainScript.py:
```diff
@@ -161,13 +168,35 @@ def check_str_is_None(string_in: str):
+            raise argparse.ArgumentError("overlapping options provided (--forcing-data-path and --exp-id)")
+
+        # got run_id and exp_id: try to use
+        # TODO we don't actually use exp_id
```
Also, if exp_id is not used, it could be removed from line 181

raehik replied:
Removed `exp_id` from CLI opts.

MarionBWeinzierl reviewed (changes requested):
Looks good, just maybe a bit of clean-up needed (check whether we can remove exp_id, remove todos).

raehik commented:
Cleaned up my to-dos, removed `--exp_id`. It doesn't get used due to how MLflow searches through its own runs. (Maybe it was required in an earlier version.)

I'm trying to fix up MLflow so that both options are available. Having issues with `None` in MLproject being parsed as a string...

raehik commented:
The `MLproject` file is restrictive, so we need a silly workaround to make it work. @MarionBWeinzierl how is this?

MarionBWeinzierl commented on src/gz21_ocean_momentum/trainScript.py:
```diff
@@ -161,13 +165,39 @@ def check_str_is_None(string_in: str):
+
+    if params.forcing_data_path is not None and params.forcing_data_path != "None":
+        # got only direct path: use
+        return params.forcing_data_path
```
Just so that I am sure to understand: In this case we do not need to bother with mlflow, right?

raehik replied:
In this case (where `--forcing-data-path` is provided), MLflow isn't used to locate the forcing data. You can run the script with or without MLflow regardless of CLI options used.
